core/src/main/scala/flow/FlowBuilder.scala

```diff
@@ -159,6 +159,7 @@ case class FlowBuilder[I, T, Ctx](
           case _ =>
             Try(f(successes.map(ir => (ir.value.get, ir.context, ir.metadata))).zip(successes).map {
               case (tryResult, FlowSuccess(_, context, metadata)) => FlowResult[U, Ctx](tryResult, context, metadata)
+              case _                                              => throw new Exception("Unexpected non-success FlowResult")
```

Comment:
exhaustiveness check is requiring to check against `FlowFailure` or other possible matches. This shouldn't be possible since we are already filtering non-success FlowResults out, but for it's good for completeness